### PR TITLE
build: Release v24.0

### PR DESCRIPTION
**Pull request by sboeuf**

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -203,6 +212,79 @@
...
+to a bypass domain. This feature is particularly helpful for booting a VM with
+guest software which doesn't support `virtio-iommu` but still need to access
+the device. Now that Cloud Hypervisor supports this feature, it can boot a VM
+with Rust Hypervisor Firmware or OVMF even if the `virtio-block` device exposing
+the disk image is placed behind a virtual IOMMU.
+
+### Ensure Identifiers Uniqueness
+
+Multiple checks have been added to the code to prevent devices with identical
+identifiers to be created, and therefore avoid unexpected behaviors at boot or
```
Comment (rbradford) [inline release-notes.md:232]: s/to be created/from being created/

File: release-notes.md [hunk 2]
```diff
@@ -203,6 +212,79 @@
...
+whenever a device was hot plugged into the VM.
+
+### Sparse Mmap support
+
+Adding sparse mmap support to both VFIO and vfio-user devices. This allows the
```
Comment (rbradford) [inline release-notes.md:237]: Use the past tense: "Adding sparse mmap support to both VFIO and vfio-user devices" => "Sparse mmap support has been added to both ..."

File: release-notes.md [hunk 3]
```diff
@@ -203,6 +212,79 @@
...
+device regions that are not fully mappable to be partially mapped. And the more
+a device region can be mapped into the guest address space, the less VM exits
```
Comment (rbradford) [inline release-notes.md:239]: s/less/fewer/

File: release-notes.md [hunk 4]
```diff
@@ -203,6 +212,79 @@
...
+will be generated when this device is accessed. This directly impacts the
+performance related to this device.
+
+### Expose Platform Serial Number
+
+Adding a new `serial_number` option to `--platform`, allowing a user to set a
```
Comment (rbradford) [inline release-notes.md:245]: s/Adding a new `serial_number` option to `--platform`/A new `serial_number` option has been added .../

File: release-notes.md [hunk 5]
```diff
@@ -203,6 +212,79 @@
...
+* Reject virtio-mem resize if device is not activated (#4003)
+* Fix potential mmap leaks from VFIO/vfio-user MMIO regions (#4069)
+* Fix algorithm finding HOB memory resources (#3983)
+
+### Notable Improvements
+
+* Refactor interrupt handling (#4083)
+* Load kernel asynchronously (#4022)
+* Only create ACPI memory manager DSDT when resizable (#4013)
+
```
Comment (rbradford) [inline release-notes.md:262]: Include a section for the deprecations as they will be removed in the next release cycle.

File: release-notes.md [hunk 6]
```diff
@@ -203,6 +212,79 @@
...
+### Contributors
```
Comment (rbradford) [inline release-notes.md:263]: Add a section "New on the Website" and reference your new blog post.

**Review by rbradford — APPROVED** [review]

**Comment by rbradford** [comment]
I forgot to suggest creating and pushing a v24.0-test tag to check the release actions (and then deleting it). This is helpful since the release action can sometimes go wrong.

**Comment by sboeuf** [comment]
> I forgot to suggest creating and pushing a v24.0-test tag to check the release actions (and then deleting it). This is helpful since the release action can sometimes go wrong.

It did go wrong :-/
That's because the toolchain 1.60 isn't installed.
```
/home/runner/.cargo/bin/cargo +1.60 build --all --release --target=x86_64-unknown-linux-gnu
error: toolchain '1.60-x86_64-unknown-linux-gnu' is not installed
Error: The process '/home/runner/.cargo/bin/cargo' failed with exit code 1
```

**Pull request merged**